server/index.js

```diff
@@ -53,6 +53,106 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const { cartId } = req.session;
+  if (cartId) {
+    const params = [cartId];
+    const cartItems = `
+      select "c"."cartItemId",
+             "c"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+       where "c"."cartId" = $1
+    `;
+    db.query(cartItems, params)
+      .then(result => res.json(result.rows))
+      .catch(err => next(err));
+  } else {
+    res.json([]);
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const { productId } = req.body;
+  if (!parseInt(productId, 10) || Math.sign(productId) !== 1) {
+    next(new ClientError('productId must be a positive integer', 400));
+    return;
+  }
+
+  const params = [productId];
+  const priceCheck = `
+    select "price"
+      from "products"
+     where "productId" = $1
+  `;
+
+  db.query(priceCheck, params)
+    .then(result => {
+      const { price } = result.rows[0];
+      if (!price) {
+        next(new ClientError(`unable to locate productId ${productId}`, 400));
```

Comment:
Here you need to `throw` the error instead of passing it to `next()` to short-circuit the promise chain to the `.catch`